Add instructions for jdk installed via brew (#9192)
duckdb/duckdb: tools/jdbc/README.md

It's required to have a JDK installed to build.

> If the build can't find the env variable `$JAVA_HOME` You must set `$JAVA_HOME` yourself.

Reviewer: ```suggestion
Make sure the `JAVA_HOME` environment variable is set.
```
